src-self-hosted/stage2.zig

```diff
@@ -1154,7 +1154,8 @@ fn enumInt(comptime Enum: type, int: c_int) Enum {
 
 fn crossTargetToTarget(cross_target: CrossTarget, dynamic_linker_ptr: *?[*:0]u8) !Target {
     var info = try std.zig.system.NativeTargetInfo.detect(std.heap.c_allocator, cross_target);
-    if (cross_target.cpu_arch == null or cross_target.cpu_model == .native) {
+    if ((cross_target.cpu_arch == null or cross_target.cpu_model == .native) and
+        (Target.current.cpu.arch != .i386 and Target.current.cpu.arch != .x86_64)) {
```

Comment:
I have an idea to simplify this logic: let's return `error.Unimplemented` from `NativeTargetInfo.detect` from the architectures that are not yet implemented, and then this logic can be simplified to running the LLVM detection if that error is returned.

Reply:
I'm not sure about this one. The only portion of `detect` that failed is the CPU detection, and the rest of the LLVM detection code relies on the rest of `info` being valid.

Reply:
> The only portion of `detect` that failed is the CPU detection, and the rest of the LLVM detection code relies on the rest of `info` being valid.

Ahhh good point. I forgot about that.

> put a flag in `NativeTargetInfo`, such as `detected_native_cpu` or something like that.

I think this will be worth it, because it will help future contributors to not accidentally introduce a mismatch between the logic in one place and another.